requirements/base.txt

```diff
@@ -1,17 +1,16 @@
 # web application
-Django==2.2.16
+Django==3.1.2
 django-crispy-forms==1.9.1
-django-filter==2.2.0
-django-redis==4.11.0
-django-reversion==3.0.7
-django-selectable==1.2.1
+django-filter==2.4.0
+django-redis==4.12.1
+django-reversion==3.0.8
+django-selectable==1.3.0
 django-taggit==1.2.0
 django-treebeard==4.3.1
 django-webpack-loader==0.7.0
-djangorestframework==3.11.0
+djangorestframework==3.11.1
```

Comment:
why not 3.12.1? https://pypi.org/project/djangorestframework/#history

Reply:
3.11.1 provided support for Django 3.1, and 3.12.0 has a much larger [change log](https://www.django-rest-framework.org/community/release-notes/#3120). I just didn't want to fight a two front war in case 3.12.0 breaks anything, but now that everything else is fixed I can upgrade that too.